pkg/artifactory/resource_artifactory_ldap_group_setting.go

```diff
@@ -51,10 +54,30 @@ func resourceArtifactoryLdapGroupSetting() *schema.Resource {
 			Description:      `(Required) The LDAP setting you want to use for group retrieval.`,
 		},
 		"group_base_dn": {
-			Type:             schema.TypeString,
-			Optional:         true,
-			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
-			Description:      `(Optional) A search base for group entry DNs, relative to the DN on the LDAP server’s URL (and not relative to the LDAP Setting’s “Search Base”). Used when importing groups.`,
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateDiagFunc: func(value interface{}, path cty.Path) diag.Diagnostics {
+				v, ok := value.(string)
+				if !ok {
+					return diag.Errorf("expected type of %q to be string", value)
+				}
+				if v == "" {
+					return nil
+				}
+				dnPat := regexp.MustCompile(`(?i)^\s*(OU|UID|DC|CN)=(.+)`)
```

Comment:
You could literally remove all this and have a regex like:

```go
regex.MustCompile(`(?i)^\s*(OU|UID|DC|CN)=(.+)|?`)
```
@alexhung ?

Reply:
Also, is the string always UPPERCASE? If not, the regex needs to account for that.

Reply:
(?i)in the pattern handles the case ignore setting.